crates/kitsune2/tests/integration.rs

```diff
@@ -444,3 +446,302 @@ async fn shutdown_space() {
+                    timeout_s: 5,
+                    webrtc_connect_timeout_s: 3,
+                    ..Default::default()
+                },
+            })
+            .unwrap();
+    }
+    kitsune_builder
+        .config
+        .set_module_config(&K2GossipModConfig {
+            k2_gossip: K2GossipConfig {
+                initiate_interval_ms: 1000,
+                min_initiate_interval_ms: 100,
+                initiate_jitter_ms: 100,
+                round_timeout_ms: 10_000,
+                ..Default::default()
+            },
+        })
+        .unwrap();
+
+    // Build should succeed.
+    assert!(kitsune_builder.build().await.is_ok());
+}
+
+#[tokio::test]
+async fn test_space_should_not_start_without_bootstrap_url_configured() {
+    enable_tracing();
+
+    // Build Kitsune2 normally, but DO NOT set any bootstrap module config.
+    let kitsune_builder = default_builder().with_default_config().unwrap();
+
+    #[cfg(any(
+        feature = "transport-tx5-backend-libdatachannel",
+        feature = "transport-tx5-backend-go-pion",
+        feature = "transport-tx5-datachannel-vendored"
+    ))]
+    {
+        let signal_server = SbdServer::new(Arc::new(sbd_server::Config {
+            bind: vec!["127.0.0.1:0".to_string()],
+            ..Default::default()
+        }))
+        .await
+        .unwrap();
+        let signal_server_url =
+            format!("ws://{}", signal_server.bind_addrs()[0]);
+        kitsune_builder
+            .config
+            .set_module_config(&Tx5TransportModConfig {
+                tx5_transport: Tx5TransportConfig {
+                    server_url: signal_server_url.to_owned(),
+                    signal_allow_plain_text: true,
+                    timeout_s: 5,
+                    webrtc_connect_timeout_s: 3,
+                    ..Default::default()
+                },
+            })
+            .unwrap();
+    }
+    kitsune_builder
+        .config
+        .set_module_config(&K2GossipModConfig {
+            k2_gossip: K2GossipConfig {
+                initiate_interval_ms: 1000,
+                min_initiate_interval_ms: 100,
+                initiate_jitter_ms: 100,
+                round_timeout_ms: 10_000,
+                ..Default::default()
+            },
+        })
+        .unwrap();
+
+    // Build should succeed.
+    let kitsune = kitsune_builder.build().await.expect("Build Kitsune2");
+
+    // register handler
+    let kitsune_handler = Arc::new(TestKitsuneHandler);
+    kitsune
+        .register_handler(kitsune_handler.clone())
+        .await
+        .expect("Register handler");
+
+    // Creating a space MUST fail because there is no bootstrap config.
+    let result = kitsune.space(TEST_SPACE_ID, None).await;
+
+    assert!(
+        result.is_err(),
+        "Expected creating a space to fail when no bootstrap URL is configured"
+    );
+}
+
+#[tokio::test]
+async fn test_should_override_bootstrap_url_per_space() {
```

Comment:
These next ones I don't mind too much whether it's an integration test or a unit test. It probably still could be a unit test and not use the real transport, I think?